### PR TITLE
[PlacementGroup]Add node manager test framework

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Add node manager test framework.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
This is really nice!

**Comment by rkooo567** [comment]
Btw, the test fails. Can you fix it before I merge the PR?

**Comment by ffbin** [comment]
> Btw, the test fails. Can you fix it before I merge the PR?

I will fix it, thanks.

File: src/ray/raylet/node_manager_test.cc
Comment (rkooo567) [inline src/ray/raylet/node_manager_test.cc:84]: Why don't we mock out this class? Is it because code is not written in a way that it is unit-testable?

Comment (ffbin) [reply src/ray/raylet/node_manager_test.cc:84]: I tried mocking it at first, but its constructor couldn't mock it. So most of UT's logic is for its constructor to execute successfully. Later, if the test needs, we can mock it.

**Comment by ffbin** [comment]
@rkooo567 I've fixed the test, thanks.

File: src/ray/raylet/node_manager_test.cc
Comment (rkooo567) [reply src/ray/raylet/node_manager_test.cc:84]: I think we don't need to test anything else other than the placement group logic, so it should be fine! (since the new scheduler will be UT'ed and object manager also will be UT'ed in the future. The only thing we need to test is the idempotency of bundle requests.)

**Comment by rkooo567** [comment]
Are you going to add tests in a separate PR?

Btw, @stephanie-wang We added this in order to unit test placement group idempotency.

**Comment by ffbin** [comment]
> Are you going to add tests in a separate PR?
> 
> Btw, @stephanie-wang We added this in order to unit test placement group idempotency.

Yeah, This PR just builds the test framework, and we will add specific UT separately later.

**Review by rkooo567 — APPROVED** [review]

**Comment by ffbin** [comment]
Hi @rkooo567 , Can this PR be merged?

**Comment by rkooo567** [comment]
Can you merge the latest master (to see if doc testing failure is unrelated) Also, I'd like to wait for @stephanie-wang's review!

**Comment by ffbin** [comment]
> Can you merge the latest master (to see if doc testing failure is unrelated) Also, I'd like to wait for @stephanie-wang's review!

Ok, i have merge the latest master, thanks.

**Comment by rkooo567** [comment]
cc @stephanie-wang I will just merge the PR first. Please feel free to let us know if you have any concern!

**Pull request merged**
